core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -750,10 +780,18 @@ private void initializeLoader(StoreConfiguration cfg, CacheLoader loader, Initia
          storesMutex.writeLock().lock();
          try {
             loaders.add(loader);
+
+            long interval = configuration.persistence().availabilityInterval();
+            ConnectableObservable<StoreAvailability> observable = Observable.interval(interval, TimeUnit.MILLISECONDS, Schedulers.from(persistenceExecutor))
+                  .map(t -> new StoreAvailability(loader, loader.isAvailable()))
+                  .doOnError(err -> log.debugf("Error encountered when calling isAvailable on %s: %s", loader, err))
+                  .retry() // Swallow any exceptions so that a call is made to isAvailable on the next interval
+                  .publish();
```

Comment:
I am not sold on using a `ConnectableObservable`. We only have a single subscriber and we don't ever invoke `connect. You should be able to just remove this line.